benchmark/sirun/initialize/init.js

```diff
@@ -0,0 +1,78 @@
+const { Rewriter } = require('../../../main')
+const csiMethods = [
+  { src: 'concat' },
+  { src: 'join' },
+  { src: 'parse' },
+  { src: 'plusOperator', operator: true },
+  { src: 'random' },
+  { src: 'replace' },
+  { src: 'slice' },
+  { src: 'substr' },
+  { src: 'substring' },
+  { src: 'toLowerCase', dst: 'stringCase' },
+  { src: 'toUpperCase', dst: 'stringCase' },
+  { src: 'tplOperator', operator: true },
+  { src: 'trim' },
+  { src: 'trimEnd' },
+  { src: 'trimStart', dst: 'trim' },
+  { src: 'eval', allowedWithoutCallee: true }
+]
+
+const orchestrion = `version: 1
+dc_module: dc-polyfill
+instrumentations:
+  - module_name: "@langchain/core"
+    version_range: ">=0.1.0"
+    file_path: dist/runnables/base.js
+    function_query:
+      name: invoke
+      type: method
+      kind: async
+      class: RunnableSequence
+    operator: tracePromise
+    channel_name: "RunnableSequence_invoke"
+  - module_name: "@langchain/core"
+    version_range: ">=0.1.0"
+    file_path: dist/runnables/base.js
+    function_query:
+      name: batch
+      type: method
+      kind: async
+      class: RunnableSequence
+    operator: tracePromise
+    channel_name: "RunnableSequence_batch"
+  - module_name: "@langchain/core"
+    version_range: ">=0.1.0"
+    file_path: dist/language_models/chat_models.js
+    function_query:
+      name: generate
+      type: method
+      kind: async
+      class: BaseChatModel
+    operator: tracePromise
+    channel_name: "BaseChatModel_generate"
+  - module_name: "@langchain/core"
+    version_range: ">=0.1.0"
+    file_path: dist/language_models/llms.js
+    function_query:
+      name: generate
+      type: method
+      kind: async
+    operator: tracePromise
+    channel_name: "BaseLLM_generate"
+  - module_name: "@langchain/core"
+    version_range: ">=0.1.0"
+    file_path: dist/embeddings.js
+    function_query:
+      name: constructor
+      type: method
+      kind: sync
+      class: Embeddings
+    operator: traceSync
+    channel_name: "Embeddings_constructor"`
+
+// eslint-disable-next-line no-new
+new Rewriter({
+  csiMethods,
+  orchestrion
+})
```

Comment:
I believe this is meant to measure the constructor performance, not the time the process needs to start up?

In that case, I would run the instance creation many times to minimize the overhead of starting the process to a bare minimum (right now it likely has a dominant part of the overall runtime) and to reduce the flakiness (running this 100 times is actually a very low iteration count for stable results).

Could we also do _something_ with the instance? It might otherwise be optimized away. Even just accessing a property would likely be enough to overcome the compiler to do dead code elimination.

A second improvement would be to to have different csi methods or a different orchestrion part being passed through and to take turns to prevent the compiler from creating to optimized code for that one configuration.

The code would roughly look like that:

```suggestion
let count = 0
for (let i = 0; i < 1e4; i++) {
  // eslint-disable-next-line no-new
  for (const csiMethods of csiMethodsVariations) {
    const instance = new Rewriter({
      csiMethods,
      orchestrion
    })
    count += instance.propertyAccess ? 1 : 0 // or similar
  }
}
```

Reply:
Doing this as you said, we are not taking into account the time that it is needed to load the wasm module on require.

What do you think if I do that, but cleaning require cache and requiring it into the for?

```javascript
  let keys
  for (const csiMethods of csiMethodsVariations) {
      const { Rewriter } = require('../../../main')
	    const instance = new Rewriter({
	      csiMethods,
	      orchestrion
	    })
	    count += instance.propertyAccess ? 1 : 0 // or similar
      keys = keys || Object.keys(require.cache)
	    keys.forEach(function(key) {
        delete require.cache[key];
      });
	  }
```

Reply:
I am not able to apply this change as I want, when I delete `require.cache` some staff still in node core memory and memory goes out, making the test invalid. I'll just address your suggestion for now, measuring only the constructor.